etest_test/build_test.py

```diff
@@ -1,12 +1,236 @@
 """Tests for etest.build."""
 
+import logging
+import textwrap
+
 import click.testing
+import docker
+import pytest
 
 import etest.build as sut
+from etest import profile
+
+logger = logging.getLogger(__name__)
 
 
 def test_help() -> None:
     """Ensure etest-build --help exits successfully."""
     runner = click.testing.CliRunner()
     result = runner.invoke(sut.main, ["--help"])
+
+    assert result.exit_code == 0
+
+
+def test_nobuild() -> None:
+    """Ensure etest-build outputs 4 info lines when not building."""
+    runner = click.testing.CliRunner()
+    result = runner.invoke(sut.main, ["--nobuild", "-v", "INFO"])
+
+    assert result.exit_code == 0
+
+    out = [o for o in result.output.splitlines() if o]
+
+    assert len(out) == 4
+
+
+def test_invalid_push(capfd: pytest.CaptureFixture[str]) -> None:
+    """Ensure _push_image() fails when provided with an invalid profile."""
```

Comment:
Invalid profiles should be tested in the profile suite instead.  This suite should assume that works and mock out that behaviour if needed.

Reply:
This actually skips the profile suite since it overrides it afterwards with nonsense.

Reply:
Ok, then let's merge these test cases and see what it looks like.  Maybe I was just lost in the length of these changes.